unitylibs/core/workflow/workflow-firefly/widget.js

```diff
@@ -0,0 +1,349 @@
+    const footer = createTag('li', { class: 'drop-footer' });
+    const tipEl = this.el.querySelector('.icon-tip')?.closest('li');
+    const tipCon = createTag('div', { id: 'tip-content', class: 'tip-con', tabindex: '-1', role: 'note', 'aria-label': `${ph['placeholder-tip']} ${tipEl?.innerText}` }, '<svg><use xlink:href="#unity-info-icon"></use></svg>');
+    const tipText = createTag('span', { class: 'tip-text', id: 'tip-text' }, `${ph['placeholder-tip']}:`);
+    const tipDesc = createTag('span', { class: 'tip-desc', id: 'tip-desc' }, tipEl?.innerText || '');
+    tipCon.append(tipText, tipDesc);
+    const legalEl = this.el.querySelector('.icon-legal')?.closest('li');
+    const legalCon = createTag('div', { class: 'legal-con' });
+    const legalLink = legalEl?.querySelector('a');
+    const legalText = createTag('a', { href: legalLink?.href || '#', class: 'legal-text' }, legalLink?.innerText || 'Legal');
+    legalCon.append(legalText);
+    footer.append(tipCon, legalCon);
+    return footer;
+  }
+
+  createActBtn(cfg, cls) {
+    if (!cfg) return null;
+    const txt = cfg.innerText?.trim();
+    const img = cfg.querySelector('img[src*=".svg"]');
+    const btn = createTag('a', { href: '#', class: `unity-act-btn ${cls}`, 'daa-ll': `Generate--${this.selectedVerbType}` });
+    if (img) btn.append(createTag('div', { class: 'btn-ico' }, img));
+    if (txt) btn.append(createTag('div', { class: 'btn-txt' }, txt.split('\n')[0]));
+    this.genBtn = btn;
+    return btn;
+  }
+
+  addWidget() {
+    const interactArea = this.target.querySelector('.copy');
+    const para = interactArea?.querySelector(this.workflowCfg.targetCfg.target);
+    this.widgetWrap.append(this.widget);
+    if (para && this.workflowCfg.targetCfg.insert === 'before') para.before(this.widgetWrap);
+    else if (para) para.after(this.widgetWrap);
+    else interactArea?.appendChild(this.widgetWrap);
+  }
+
+  async loadPrompts() {
+    const { locale } = getConfig();
+    const { origin } = window.location;
+    const baseUrl = (origin.includes('.aem.') || origin.includes('.hlx.'))
+      ? `https://main--unity--adobecom.${origin.includes('.hlx.') ? 'hlx' : 'aem'}.live`
+      : origin;
+    const promptFile = locale.prefix && locale.prefix !== '/'
+      ? `${baseUrl}${locale.prefix}/unity/configs/prompt/firefly-prompt.json`
+      : `${baseUrl}/unity/configs/prompt/firefly-prompt.json`;
+    const promptRes = await fetch(promptFile);
+    if (!promptRes.ok) {
+      throw new Error('Failed to fetch prompts.');
+    }
+    const promptJson = await promptRes.json();
+    this.prompts = this.createPromptMap(promptJson?.content?.data);
+  }
+
+  async getPrompt(verb) {
+    if (!this.hasPromptSuggestions) return [];
+    try {
+      if (!this.prompts || Object.keys(this.prompts).length === 0) await this.loadPrompts();
+      return (this.prompts?.[verb] || []).filter((item) => item.prompt && item.prompt.trim() !== '');
+    } catch (e) {
+      return [];
```

Comment:
Should we log to lana here if we catch an exception here?

Reply:
Added lanalog